js_modules/dagit/packages/core/src/assets/LaunchAssetExecutionButton.tsx

```diff
@@ -60,39 +61,72 @@ type LaunchAssetsState =
 
 const countOrBlank = (k: unknown[]) => (k.length > 1 ? ` (${k.length})` : '');
 
+type Asset =
+  | {assetKey: AssetKey; partitionDefinition: {__typename: string} | null; isSource: boolean}
+  | {assetKey: AssetKey; isPartitioned: boolean; isSource: boolean};
+
+type AssetsInScope = {all: Asset[]; skipAllTerm?: boolean} | {selected: Asset[]};
+
+type LaunchOption = {assetKeys: AssetKey[]; label: string};
+
+const isAnyPartitioned = (assets: Asset[]) =>
+  assets.some(
+    (a) =>
+      ('partitionDefinition' in a && !!a.partitionDefinition) ||
+      ('isPartitioned' in a && a.isPartitioned),
+  );
+
+function optionsForButton(scope: AssetsInScope, liveDataForStale?: LiveData): LaunchOption[] {
+  // If you pass a set of selected assets, we always show just one option
+  // to materialize that selection.
+  if ('selected' in scope) {
+    const assets = scope.selected.filter((a) => !a.isSource);
+    return [
+      {
+        assetKeys: assets.map((a) => a.assetKey),
+        label: `Materialize selected${countOrBlank(assets)}${isAnyPartitioned(assets) ? '…' : ''}`,
+      },
+    ];
+  }
+
+  const options: LaunchOption[] = [];
+  const assets = scope.all.filter((a) => !a.isSource);
+
+  options.push({
+    assetKeys: assets.map((a) => a.assetKey),
+    label:
+      assets.length > 1 && !scope.skipAllTerm
```

Comment:
`skipAllTerm` is a bit of a hack, but the problem is that on the Partitions view "All" seems like it's talking about all partitions, not all assets. In this /one/ case, Josh and I want to hide All 🫣